jupyter_console/interactiveshell.py

```diff
@@ -220,6 +224,43 @@ def handle_execute_reply(self, msg_id, timeout=None):
 
             self.execution_count = int(content["execution_count"] + 1)
 
+    def handle_is_complete_reply(self, msg_id, timeout=None):
+        """
+        `complete` - code is ready to be executed
+        `incomplete` - code should prompt for another line
+        `invalid` - code will typically be sent for execution, so that the user sees the error soonest.
+        `unknown` - if the kernel is not able to determine this. The
+                    frontend should also handle the kernel not replying promptly. It
+                    may default to sending the code for execution, or it may implement
+                    simple fallback heuristics for whether to execute the code
+                    (e.g. execute after a blank line).
```

Comment:
This looks like a copy-paste from the docs. Since these statuses are handled within this function, and not exposed outside it, I don't think they need to be described in the docstring.